searx/templates/simple/preferences.html

```diff
@@ -314,9 +314,9 @@ <h1>{{ _('Preferences') }}</h1>
         <th>{{ _("Max time") }}</th>{{- "" -}}
         <th>{{ _("Reliability") }}</th>{{- "" -}}
       </tr>
-      {% for group, engines in engines_by_category[categ] | group_engines_in_tab %}
+      {% for group, group_bang, engines in engines_by_category[categ] | group_engines_in_tab %}
       {% if loop.length > 1 %}
-      <tr><th colspan="9" class="engine-group">{{_(group)}}</th></tr>
+      <tr><th colspan="9" class="engine-group">{{_(group)}}{% if group_bang %}<span class="group-bang">{{group_bang}}</span>{% endif %}</th></tr>{{- "" -}}
```

Comment:
The text  `{{group_bang}}` is not an button, so it should not be displayed as a button.

Parenthesis and italic should be enough:

Général ( _!general_ )

Reply:
> Parenthesis and italic should be enough

I think different. Before I fail the nest time: please place a patch on top with what you think what is the best.

Reply:
I've commit for this:
![image](https://user-images.githubusercontent.com/1594191/182177983-9ec1f621-b4b6-4d09-9044-221cb1e0c898.png)

But this is misleading: actually `!web` includes engines in the general, and videos categories, so if we document the bang we should list the engines too (in the tooltip).

Reply:
I miss the translation in your screenshot .. for me the translation makes sense and I want to keep it.

![image](https://user-images.githubusercontent.com/554536/182200284-55b03466-b92c-46cc-8703-4c92b262ac79.png)

![image](https://user-images.githubusercontent.com/554536/182200371-5ef6ab84-5f31-4556-8523-0c2e4427a0f3.png)

> But this is misleading: !web includes engines in general and videos categories, so if we document the bang we should list the engines too (in the tooltip).

This is a logical consequence of the two-dimensional grouping, I don't have a clue how we can display it .. the tooltip can help in WEB UI but we have the same question in the documentation : https://docs.searxng.org/admin/engines/configured_engines.html#configured-engines

TBH I think to document & display the "two-dimensional grouping" well needs more brainpower .. we can discuss in a different PR.

> The text {{group_bang}} is not an button, so it should not be displayed as a button.

I switched to the colors we already use in the online "search syntax" ..

![image](https://user-images.githubusercontent.com/554536/182203749-28bdc954-83b8-49c9-b961-f4a63a3ad7a1.png)

@dalf from my side the PR is ready to merge, what do you think?

Reply:
> I switched to the colors we already use in the online "search syntax" ..

👍 for the usage of `--color-doc-code` and `--color-doc-code-background`.

However:
* the red color has now two meanings in `/preferences`: either bang, either long response time.
* it adds more data in an already overloaded page. That's why I used tooltips. What is your opinion about tooltip?

---

We can use the blue color to display the bangs:
```css
--color-doc-code: #003;
--color-doc-code-background: #ddeaff;
```

![image](https://user-images.githubusercontent.com/1594191/182214757-9d2aa872-5dc2-4ee3-b6e2-29d7fb01ff03.png)

---

> TBH I think to document & display the "two-dimensional grouping" well needs more brainpower .. we can discuss in a different PR.

So there are two PR :-)

Rather than one PR will small steps in two different subjects (settings & display the bang in the preferences), I prefer two PR each focused on the same subject.

Reply:
> We can use the blue color to display the bangs:

I changed the colors.

> So there are two PR :-)
> Rather than one PR will small steps in two different subjects (settings & display the bang in the preferences), I prefer two PR each focused on the same subject. 

With all respect, I have a different opinion on what makes a PR worth to merge, by example:

ATM I don't have a chance to know with what bang I can make use of the "Lexika" ..

![grafik](https://user-images.githubusercontent.com/554536/182242318-38a91d60-63b4-474a-993f-70eaa7d466a5.png)

with this PR I have the information.  IMO this is a "progress" and worth to merge, even it is not perfect since there is a lot more to say about bangs .. and when we look at all the category (and category in UI) -bangs, the engine-bangs and the hundreds of external-bangs we have, than we can guess that this topic needs a detailed explanation and a good overview.  I don't have a clue how we can describe all the details and improve accessibility .. may be we add a new info page or improve the auto-completion .. IDK right now.

But having the Lexika --> !dictionaries is a good improvement to me, that is worth to merge.  SearXNG is a rolling release, there is no need to make a feature perfect in the first "release".